app/models/concerns/hyrax/collection_behavior.rb

```diff
@@ -14,8 +14,15 @@ module CollectionBehavior
     included do
       validates_with HasOneTitleValidator
       self.indexer = Hyrax::CollectionIndexer
+
+      property :collection_type_gid, predicate: ::RDF::Vocab::SCHEMA.additionalType, multiple: false do |index|
+        index.as :symbol, :stored_searchable
+      end
     end
 
+    attr_accessor :collection_type
```

Comment:
RE: validation - That makes sense.  There are some loose ends around this because the code for setting this in the controller isn't in place yet.  The load happens for edit, but is pending some work on the New Collection to incorporate the modal selection of collection type process.  And we don't have all the collection_type code done to be able to create a collection type to be selected.  Bottom line, we can't put the set and persisted test in place until some of these other parts fall into place.  I'll make a note in the main issue for this.

RE: nestable?, etc. - These methods are delegates to the collection_type.  If the collection_type doesn't exist, then teh object is in an error state.  Any defaults should be controlled by the collection_type.

RE: collection_type= - Depends on how we want to set the collection type for new collections?  Thinking through the implications...  this means that the collection_type needs to provide its gid AND the new collection modal for selecting a collection_type should have the gid for each collection_type such that when the new collection object is created, it is passed the gid from the modal.